src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -877,6 +877,11 @@ MulticopterAttitudeControl::task_main()
 					}
 				}
 
+				/* weather-vane mode: disable yaw control */
+				if(_v_att_sp.disable_mc_yaw_control == true){
+					_rates_sp(2) = _ctrl_state.yaw_rate;
```

Comment:
This is not the right way to do this. You can get into an increasing yaw rate this way. A simple wind gust would be sufficient to start the spiral of death. What you should do instead is to set the actuator control output for yaw to zero.